File: leetcode/100-1000/leetcode589.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

class Node {
public:
    int val;
    vector<Node*> children;

    Node() { }

    Node(int _val)
    {
        val = _val;
    }

    Node(int _val, vector<Node*> _children)
    {
        val = _val;
        children = _children;
    }
};

class Solution {
public:
    vector<int> preorder(Node* root)
    {
        vector<int> ans;
        anya(root, ans);
        return ans;
    }
    void anya(Node* root, vector<int>& ans)
    {
        if (root == nullptr)
            return;
        ans.push_back(root->val);
        for (auto i : root->children) {
            anya(i, ans);
        }
        return;
    }
};
int main(void)
{
    return 0;
}

File: leetcode/100-999/leetcode589.py
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


from typing import List


class Solution:
    def preorder(self, root: Node) -> List[int]:
        if not root:
            return []

        res = []

        def chaewon(cur_node: Node) -> None:
            res.append(cur_node.val)
            for child in cur_node.children:
                chaewon(child)

        chaewon(root)
        return res
